tutor4points/core/views.py

```diff
@@ -58,35 +58,59 @@ def home(request):
 
 
 # tutors handles "/tutors" endpoint
-# allows user to view all tutors that attend the same school as them
+# allows user to view all tutors that attend the same school as them and send tutor requests
 @login_required
 def tutors(request):
-    if request.method == 'GET':
-        current_user = request.user
-        users = get_user_model().objects.all().exclude(pk=current_user.id)
-        tutors = users.filter(is_tutor=True, school=current_user.school)
+    current_user = request.user
+    users = get_user_model().objects.all().exclude(pk=current_user.id)
+    tutors = users.filter(is_tutor=True, school=current_user.school)
+        
+    if request.method == 'POST':
+        form = TutorRequestForm(request.POST)
+        if(form.is_valid()):
+            tutor_instance = get_user_model().objects.get(pk=request.POST['requestedTutor']) # get tutor object
+            tutor_request = form.save(commit=False)
+            tutor_request.tutee = current_user
+            tutor_request.tutor = tutor_instance
+            tutor_request.save()
+            return redirect('requests', id=current_user.id)
```

Comment:
Rather than redirect to the Requests page, it's prob better to just keep the user on the tutors page and display a success message like the one shown on the Points page. You can include a href to the requests page in the success message. For example "Success, you have sent a tutor request to .... To view all sent requests, please go to <link>"